Wingstop_Recovery_Computer_V1/Wingstop_Recovery_Computer_V1.ino

```diff
@@ -389,137 +460,224 @@ void read_barometer(){//takes barometer reading
 
-  landing = x1; //updates landing variable for detection
-  x1 = read_altitude(); //updates altitude
-  t1 = t2;  //updates new time
+  x_previous = x_current; //updates previous position variable for landing detection
+  x_current = read_altitude(); //updates altitude
+  t_previous = t_current;  //updates new time
 }
 
 
 //***********FILE FUNCTIONS***********
 
 void open_file(){// opens the file for writing
-  myFile = SD.open("flight.txt", FILE_WRITE);
-
-  
-  if (myFile) {
-    //file opened ok
-  } else {
-    // if the file didn't open, turn on buzzer/LED to indicate file problem
-    turn_on_led(); //Turn on LED
-    turn_on_buzzer(); //Turn on buzzer
-    while(1){ 
-      //run while loop forever
-    }
-  }
+  logFile = SD.open("flight.txt", FILE_WRITE);
 }
 
 void set_header_file(){//sets headers at begining of file
-  myFile.print(F("Flight Log:\t"));
-  myFile.print(hz);
-  myFile.println(F("hz"));
-  myFile.print(F("Time:\tAlt:\tApo:\tEvents:\t"));
+  logFile.print(F("Flight Log:\t\t"));
+  logFile.print(F("SEALEVELPRESSURE:\t"));
+  logFile.print(SEALEVELPRESSURE_HPA); logFile.print(F(" [HPA]\t")); 
+  logFile.print(F("hz:\t"));
+  logFile.print(hz); logFile.print(F(" [hz]\t")); 
+  logFile.print(F("Initial Altitude:\t"));
+  logFile.print(init_altitude); logFile.print(F(" [m]\t")); 
+  logFile.print(F("TAKEOFF_ALTITUDE:\t"));
+  logFile.print(TAKEOFF_ALTITUDE); logFile.print(F(" [m]\t")); 
+  logFile.print(F("MAIN_CHUTE_ALTITUDE:\t"));
+  logFile.print(MAIN_CHUTE_ALTITUDE); logFile.print(F(" [m]\t")); 
+  logFile.print(F("DROGUE_DELAY:\t"));
+  logFile.print(DROGUE_DELAY); logFile.print(F(" [millis]\t")); 
+  logFile.print(F("MACH_DELAY:\t"));
+  logFile.print(MACH_DELAY); logFile.println(F(" [millis]\t")); 
+  
+  logFile.print(F("Time:\tAltitude:\tEvents:\t"));
+
+
+
+  
 }
 
 void close_file(){
-  myFile.close(); //closes file
+  logFile.close(); //closes file
 }
 
 void reopen_file(){//closes then reopens the file, saving data up to this point
+  if (logFile){
   close_file();//closes the file
+  }
   open_file();//opens the file
 }
 
 void log_data(){//saves current data to sd card
-  myFile.print("\n");
-  myFile.print(t2); myFile.print(F("\t"));       //logs current time
-  myFile.print(x1); myFile.print(F("\t"));       //logs current position
-  
+  if (logFile) {
+  logFile.print("\n");
+  logFile.print(t_current); logFile.print(F("\t"));       //logs current time
+  logFile.print(x_current); logFile.print(F("\t"));       //logs current position
+  }
 }
 
+int SD_findInt(const __FlashStringHelper * key) {
+  char value_string[30];
+  int value_length = SD_findKey(key, value_string);
+  return HELPER_ascii2Int(value_string, value_length);
+}
 
+float SD_findFloat(const __FlashStringHelper * key) {
+  char value_string[30];
+  int value_length = SD_findKey(key, value_string);
+  return HELPER_ascii2Float(value_string, value_length);
+}
 
+int SD_findKey(const __FlashStringHelper * key, char * value) {
```

Comment:
I didn't realize reading a config file would be such a pain!  Were all the SD_ functions tested successfully?

Reply:
Yes, they read off the config.txt file just fine.  I tested it by having it Serial print the values on startup, and they are also logged at the beginning of each flight data set.